anta/tests/hardware.py

```diff
@@ -379,3 +376,68 @@ def test(self) -> None:
         # Verify that the expected redundancy protocol configured, operational and switchover ready
         elif not command_output["switchoverReady"]:
             self.result.is_failure(f"Redundancy protocol switchover status mismatch - Expected: True Actual: {command_output['switchoverReady']}")
+
+
+class VerifyFlashUtilization(AntaTest):
+    """Verifies the free space percentage on the flash drive.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the flash utilization for the primary, and the peer supervisor (if check_peer_supervisor is True), is below the defined
+     threshold.
+    * Failure: The test will fail if the flash utilization for the primary, or for the peer supervisor (when check_peer_supervisor is True), is above the defined
+     threshold.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.hardware:
+      - VerifyFlashUtilization:
+          flash_utilization_threshold: 70
+          check_peer_supervisor: True # Optional
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["hardware"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [
+        AntaTemplate(template="{check_peer_supervisor}show file systems", revision=1),
+    ]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyFlashUtilization test."""
+
+        check_peer_supervisor: bool = False
+        """If True, extends free space verification to the backup supervisor's flash drive."""
+        flash_utilization_threshold: Percent = 70
+        """The maximum allowed percentage of flash memory utilization."""
```

Comment:
I think we usually put boolean fields last:
```suggestion
        max_utilization: Percent = 70
        """The maximum allowed percentage of flash memory utilization."""
        check_peer_supervisor: bool = False
        """If True, also verifies the peer supervisor flash drive."""
```